chain/network/src/peer_manager/tests/tier1.rs

```diff
@@ -0,0 +1,340 @@
+    assert_eq!(RoutedMessageBody::BlockApproval(want), got.body);
+}
+
+async fn test_clique(rng: &mut Rng, pms: &[&peer_manager::testonly::ActorHandler]) {
+    // Send a message over each connection.
+    for from in pms {
+        for to in pms {
+            if from.cfg.node_id() == to.cfg.node_id() {
+                continue;
+            }
+            send_tier1_message(rng, from, to).await;
+        }
+    }
+}
+
+// In case a node is its own proxy, it should advertise its address as soon as
+// it becomes a TIER1 node.
+#[tokio::test]
+async fn first_proxy_advertisement() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+    let pm = peer_manager::testonly::start(
+        clock.clock(),
+        near_store::db::TestDB::new(),
+        chain.make_config(rng),
+        chain.clone(),
+    )
+    .await;
+    let chain_info = peer_manager::testonly::make_chain_info(&chain, &[&pm]);
+    tracing::info!(target:"test", "set_chain_info()");
+    // TODO(gprusak): The default config constructed via chain.make_config(),
+    // currently returns a validator config with its own server addr in the list of TIER1 proxies.
+    // You might want to set it explicitly within this test to not rely on defaults.
+    pm.set_chain_info(chain_info).await;
+    let got = pm.tier1_advertise_proxies(&clock.clock()).await;
+    tracing::info!(target:"test", "awaiting for Tier1AdvertiseProxies");
+    assert_eq!(
+        got[0].proxies,
+        vec![PeerAddr { peer_id: pm.cfg.node_id(), addr: pm.cfg.node_addr.unwrap() }]
+    );
+}
+
+#[tokio::test]
+async fn direct_connections() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    let mut pms = vec![];
+    for _ in 0..5 {
+        pms.push(
+            peer_manager::testonly::start(
+                clock.clock(),
+                near_store::db::TestDB::new(),
+                chain.make_config(rng),
+                chain.clone(),
+            )
+            .await,
+        );
+    }
+    let pms: Vec<_> = pms.iter().collect();
+
+    tracing::info!(target:"test", "Connect peers serially.");
+    for i in 1..pms.len() {
+        pms[i - 1].connect_to(&pms[i].peer_info(), tcp::Tier::T2).await;
+    }
+
+    tracing::info!(target:"test", "Set chain info.");
+    let chain_info = peer_manager::testonly::make_chain_info(&chain, &pms[..]);
+    for pm in &pms {
+        pm.set_chain_info(chain_info.clone()).await;
+    }
+    tracing::info!(target:"test", "Establish connections.");
+    establish_connections(&clock.clock(), &pms[..]).await;
+    tracing::info!(target:"test", "Test clique.");
+    test_clique(rng, &pms[..]).await;
+}
+
+#[tokio::test]
+async fn proxy_connections() {
```

Comment:
please add a comment describing the test setup

Reply:
done